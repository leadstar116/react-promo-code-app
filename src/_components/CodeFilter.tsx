import React from 'react'
import { Button, FormControl, InputGroup } from 'react-bootstrap'

type Props = {
  searchString: string,
  setSearchString: (name: string) => void
}

const CodeFilter = (props: Props) => {
  return (
    <div className="code-filter-wrapper">
      <label htmlFor="code-filter">FILTER</label>
      <InputGroup className="mr-sm-2 search-input" >
        <FormControl
          id="code-filter"
          type="text"
          placeholder="Search"
          className={props.searchString
            ? "mr-sm-2 border border-danger"
            : "mr-sm-2"
          }
          value={props.searchString}
          onChange={ (e) => props.setSearchString(e.target.value) }
        />
        <Button
          className="btn-reset-filter"
          onClick={ () => props.setSearchString('') }
        >
          Reset
        </Button>
      </InputGroup>
    </div>
  )
}

export default CodeFilter