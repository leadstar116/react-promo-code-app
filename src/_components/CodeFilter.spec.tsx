import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CodeFilter from './CodeFilter'


describe('CodeFilter Component', () => {
	let component = renderer.create(
		<BrowserRouter>
			<CodeFilter searchString='' setSearchString={(name: string) => {}}/>
		</BrowserRouter>
	);

	it('should render with given props', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
})