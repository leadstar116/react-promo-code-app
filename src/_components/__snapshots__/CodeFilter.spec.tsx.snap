// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeFilter Component should render with given props 1`] = `
<div
  className="code-filter-wrapper"
>
  <label
    htmlFor="code-filter"
  >
    FILTER
  </label>
  <div
    className="mr-sm-2 search-input input-group"
  >
    <input
      className="mr-sm-2 form-control"
      id="code-filter"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <button
      className="btn-reset-filter btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;
